pkg/prometheus/promcfg.go

```diff
@@ -116,6 +116,26 @@ func addTLStoYaml(cfg yaml.MapSlice, namespace string, tls *v1.TLSConfig) yaml.M
 	return cfg
 }
 
+func addAuthorizationToYaml(cfg yaml.MapSlice, assetStoreKey string, store *assets.Store, auth *v1.Authorization) yaml.MapSlice {
+	if auth != nil {
```

Comment:
Might be better to remove the indentation with an early return
```
if auth == nil {
  return cfg
}
```

Reply:
yeah makes sense, especially with the version check.